smatch/tests/test_matcher.py

```diff
@@ -204,3 +206,76 @@ def test_sphdist():
     d = sphdist([[10, 10], [10, 10]], [[20, 20], [21, 21]], [[10, 10]], [[21, 21.5]])
     assert np.shape(d) == (2, 2)
     assert np.allclose(d, [[1, 1.5], [0, 0.5]], atol=2e-6)
+
+
+def test_matcher_self_radius():
+    ra, dec = _gen_sphere_pts(50, 4543)
+    mch = Matcher(ra, dec)
+
+    idx = mch.query_self(6e4/3600)
+    assert any(len(_idx) > 1 for _idx in idx)
+
+    for ic in range(ra.shape[0]):
+        idxc = []
+        for ip in range(ra.shape[0]):
+            sep = sphdist(ra[ic], dec[ic], ra[ip], dec[ip])
+            if sep < 6e4/3600:
+                idxc.append(ip)
+        assert set(idxc) == set(idx[ic])
```

Comment:
@erykoff I checked explicitly and the old code fails this simple test and so is wrong.